File: skills/playwright/src/commands/screenshot.rs
use chromiumoxide::page::ScreenshotParams;

use crate::browser::BrowserSession;

pub async fn screenshot(
    session: &mut BrowserSession,
    output: Option<&str>,
    full_page: bool,
) -> anyhow::Result<serde_json::Value> {
    let timestamp = std::time::SystemTime::now()
        .duration_since(std::time::SystemTime::UNIX_EPOCH)
        .unwrap_or_default()
        .as_millis();

    let path = output
        .map(|s| s.to_string())
        .unwrap_or_else(|| format!("/tmp/pw-screenshot-{timestamp}.png"));

    // Ensure parent directory exists
    if let Some(parent) = std::path::Path::new(&path).parent() {
        tokio::fs::create_dir_all(parent).await?;
    }

    let params = ScreenshotParams::builder()
        .full_page(full_page)
        .build();

    let bytes = session.page.screenshot(params).await?;
    tokio::fs::write(&path, &bytes).await?;

    Ok(serde_json::json!({
        "path": path,
        "url": session.url().await,
    }))
}

File: skills/playwright/src/commands/screenshot.ts
import { homedir } from "node:os";
import { join } from "node:path";
import { getBrowser } from "../browser.js";

export interface ScreenshotResult {
	success: boolean;
	path: string;
	url: string;
	timestamp: string;
	error?: string;
}

export interface ScreenshotOptions {
	output?: string;
	fullPage?: boolean;
}

export async function screenshot(
	options: ScreenshotOptions = {},
): Promise<ScreenshotResult> {
	try {
		const { page } = await getBrowser();

		const outputPath =
			options.output ||
			join(
				homedir(),
				".mini-claw",
				"playwright",
				`screenshot-${Date.now()}.png`,
			);

		await page.screenshot({
			path: outputPath,
			fullPage: options.fullPage ?? false,
		});

		return {
			success: true,
			path: outputPath,
			url: page.url(),
			timestamp: new Date().toISOString(),
		};
	} catch (err) {
		const error = err instanceof Error ? err.message : String(err);
		return {
			success: false,
			path: options.output || "",
			url: "",
			timestamp: new Date().toISOString(),
			error,
		};
	}
}
